src/java.base/share/classes/java/lang/reflect/code/type/MethodRef.java

```diff
@@ -69,8 +70,10 @@ public sealed interface MethodRef extends TypeVarRef.Owner permits MethodRefImpl
 
     // Factories
 
-    static MethodRef method(Method m) {
-        return method(m.getDeclaringClass(), m.getName(), m.getReturnType(), m.getParameterTypes());
+    static MethodRef method(Executable e) {
```

Comment:
I generalized this, as we needed some way to create a `MethodRef` from a constructor, but also realizing that `MethodRef` is used for both methods and constructors, so this generalization seems to make sense (if not, I can hide the new functionality).

Reply:
I think that's good enough for now. We could later introduce an sealed ExecutableRef with MethodRef and ConstructorRef.